emperor/util.py

```diff
@@ -652,3 +654,46 @@ def guess_coordinates_files(dir_path):
         coord_fps.append(filepath)
 
     return coord_fps
+
+
+def validate_and_process_custom_axes(mf, custom_axes):
+    """Validate and process mapping file for custom axes
+
+    Parameters
+    ----------
+    mf : pd.DataFrame
+        The sample metadata.
+    custom_axes : list of str
+        The custom axes to extract from the metadata.
+
+    Returns
+    -------
+    pd.DataFrame
+        A resulting DataFrame with the custom axes as numeric types.
+
+    Raises
+    ------
+    KeyError
+        When the category names in `custom_axes` are not present in `mf`.
+    ValueError
+        When there are non-numeric values in the categories selected by
+        `custom_axes`.
+    """
+    # avoid side-effects
+    mf = mf.copy()
+
+    headers = [mf.index.name] + mf.columns.tolist()
+    missing_headers = set(custom_axes).difference(set(headers))
+
+    if missing_headers:
+        raise KeyError("One or more headers are not present in the "
```

Comment:
Isn't this situation, one or more requested custom axes is not present in the sample metadata?